eng/pipelines/templates/steps/vcpkg.yml

```diff
@@ -1,146 +1,52 @@
-    displayName: MacOS - Set VCPKG_INSTALLATION_ROOT
+  - pwsh: |
+      Write-Host "##vso[task.prependpath]$(Agent.TempDirectory)/vcpkg"
+      Write-Host "##vso[task.setvariable variable=VCPKG_INSTALLATION_ROOT;]$(Agent.TempDirectory)/vcpkg"
+      Write-Host "##vso[task.setvariable variable=VcpkgRef;]$(Get-Content eng/vcpkg-ref.txt)"
+    displayName: Set Vcpkg Variables
     condition: >-
       and(
         succeeded(),
-        not(eq(variables['${{ parameters.DependenciesVariableName }}'], '')),
-        contains(variables['OSVmImage'], 'macOS')
+        not(eq(variables['${{ parameters.DependenciesVariableName }}'], ''))
       )
 
-  # Set MacOS specific build environment features for vcpkg
-  - task: Bash@3
+  - task: Cache@2
     inputs:
-      targetType: inline
-      script: |
-        sudo xcode-select --switch /Applications/Xcode_11.3.1.app
-        echo "xcode path:"
-        sudo xcode-select --print-path
-
-        # Install gcc 9
-        brew install gcc@9
-        gcc --version
-    displayName: MacOS - Set tools versions
+      key: >-
+        $(Agent.JobName)
+        | "$(VcpkgRef)"
+        | $(Agent.Os)
+        | $(${{ parameters.DependenciesVariableName }})
+      path: $(Agent.TempDirectory)/vcpkg
+      cacheHitVar: VcpkgRestoredFromCache
+    displayName: Vcpkg Cache
     condition: >-
       and(
         succeeded(),
         not(eq(variables['${{ parameters.DependenciesVariableName }}'], '')),
-        contains(variables['OSVmImage'], 'macOS')
+        not(eq(variables['Skip.VcpkgCache'], 'true'))
       )
 
-  # Clear the vcpkg folder so the cache can successfully deploy to that location
-  - bash: |
-      id
-      sudo rm -rf $(VCPKG_INSTALLATION_ROOT)
-      sudo mkdir -p $(VCPKG_INSTALLATION_ROOT)
-      sudo chown `id --name -u`.`id --name -g` $(VCPKG_INSTALLATION_ROOT)
-    displayName: Linux - Clear vcpkg folder
-    condition: >-
-      and(
-        succeeded(),
-        not(eq(variables['${{ parameters.DependenciesVariableName }}'], '')),
-        contains(variables['OSVmImage'], 'ubuntu')
-      )
-
-  # Install vcpkg on MacOS
-  - task: Bash@3
+  - task: PowerShell@2
     inputs:
-      targetType: inline
-      workingDirectory: $(Agent.TempDirectory)
-      script: |
-        git clone https://github.com/Microsoft/vcpkg.git
-        cd vcpkg
-        git rev-parse --verify HEAD
-
-        # Make sure to fetch all the tags to avoid failures on checkout.
-        git fetch --tags
-
-        git checkout ${{ parameters.VcpkgVersion }}
-        git status
-
-        ./bootstrap-vcpkg.sh
-
-        # Validate that vcpkg bootstrap succeeded
-        ./vcpkg version
-        if [ $? -ne 0 ]
-        then
-          echo "./bootstrap-vcpkg.sh FAILED"
-          exit 1
-        fi
+      targetType: filePath
+      filePath: eng/scripts/vcpkg.ps1
+      arguments: >-
+        -Ref $(VcpkgRef)
+        -Dependencies "$(${{ parameters.DependenciesVariableName }})"
```

Comment:
Is there ever going to be a case where different steps pass different Dependencies? That seems like it will really mess up the caching system. For example we may end up with X versions of vcpkg each with a slightly different dependency set.